lib/spack/spack/spec.py

```diff
@@ -4995,53 +4995,74 @@ def __missing__(self, key):
 
 #: These are possible token types in the spec grammar.
 HASH, DEP, VER, COLON, COMMA, ON, OFF, PCT, EQ, ID, VAL, FILE = range(12)
+#  0,   1,   2,     3,     4,  5,   6,   7,  8,  9,  10,   11
 
 #: Regex for fully qualified spec names. (e.g., builtin.hdf5)
 spec_id_re = r"\w[\w.-]*"
 
 
+_filename_re_base = r"[/\w.-]*/[/\w/-]+\.(yaml|json)[^\b]*"
+_windows_filename_prefix = r"([A-Za-z]:)*?"
+# Spec strings require posix-style paths on Windows
+# because the result is later passed to shlex.
+_windows_filename_re = _windows_filename_prefix + _filename_re_base
```

Comment:
I haven't changed any of the actual strings here whatsoever, I've just moved them to the top level.